engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -168,8 +171,10 @@ class WorkflowManagerActor(backend: Backend, config: Config)
 
   when (Running) {
     case Event(SubmitWorkflow(source), _) =>
-      val updatedData = submitWorkflow(source, replyTo = Option(sender), id = None)
-      stay() using updatedData
+      submitWorkflow(source, replyTo = Option(sender), id = None) onSuccess {
+        case updatedData: WorkflowManagerData => self ! UpdateWorkflowManagerState(updatedData)
+      }
+      stay()
```

Comment:
We are only appending the info to the state data (via messages, which this actor will guarantee handling in a serialized way), via `add` method. I don't think any overwriting is being performed   (or attempted currently). Or have I got it terribly wrong?